### PR TITLE
Guestbook

### PR DESCRIPTION
**Pull request by epuzio**

### CODE REVIEW

File: src/_GuestBook/GuestBook.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+import javax.swing.text.JTextComponent;
+
+public class GuestBook implements ActionListener {
+	JPanel panel = new JPanel();
+	JFrame frame = new JFrame();
+	JLabel checkInText = new JLabel();
```
Comment (IsisJ) [inline src/_GuestBook/GuestBook.java]: For your JLabel, you can set its text (as a string) as one of its parameters.

File: src/_GuestBook/GuestBook.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	ArrayList <Guest> guestList= new ArrayList <> ();
+
+	public void actionPerformed(ActionEvent e) {
+		JButton buttonPressed = (JButton) e.getSource();
+		
+		
+		if(buttonPressed == guestButton) {
+			RegularGuest regularGuest = new RegularGuest(guestNameField.getText());
+			guestList.add(regularGuest);
+			System.out.println(guestList);
```
Comment (IsisJ) [inline src/_GuestBook/GuestBook.java]: Maybe you don't need to print out the list to the console anymore?

File: src/_GuestBook/RegularGuest.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package _GuestBook;
+
+public class RegularGuest implements Guest {
+	RegularGuest(String name){
+		this.name = name;
+	}
+	String name;
```
Comment (IsisJ) [inline src/_GuestBook/RegularGuest.java]: Ordering files: maybe place all instance variable(s) at the top, before the constructor

File: src/_GuestBook/RegularGuest.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	public String toString() {
+		return name;
+	}
+
+	@Override
+	public String getListing() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
```
Comment (IsisJ) [inline src/_GuestBook/RegularGuest.java:16]: If you're all done, go ahead and remove the "@Override" and the todo comment!

File: src/_GuestBook/RegularGuest.java (hunk 3 above)
Comment (IsisJ) [reply src/_GuestBook/RegularGuest.java]: Also, I think it's safe to make the name variable private

File: src/_GuestBook/GuestBook.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	
+	public static void main(String[] args) {
+		GuestBook guestbook = new GuestBook();
+		guestbook.createUI();
+	}
+	
+	public String buildNiceList(ArrayList list) {
+		return "test";
+		
+	}
```
Comment (IsisJ) [inline src/_GuestBook/GuestBook.java:75]: Do you still use this method?

Comment (IsisJ) [reply src/_GuestBook/GuestBook.java:75]: Formatting is consistent 🐐

File: src/_GuestBook/GuestBook.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+public class GuestBook implements ActionListener {
+	JPanel panel = new JPanel();
+	JFrame frame = new JFrame();
+	JLabel checkInText = new JLabel();
+	JButton guestButton = new JButton();
+	JButton VIPButton = new JButton();
+	JTextField guestNameField = new JTextField(20);
+	JButton seeAllButton = new JButton();
+
```
Comment (IsisJ) [inline src/_GuestBook/GuestBook.java:22]: All instance variables in one spot! 💯

File: src/_GuestBook/GuestBook.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+		
+		if(buttonPressed == guestButton) {
+			RegularGuest regularGuest = new RegularGuest(guestNameField.getText());
+			guestList.add(regularGuest);
+			System.out.println(guestList);
+		}
+		
+		if(buttonPressed == VIPButton) {
+			VIPGuest vipGuest = new VIPGuest("VIP: " +guestNameField.getText());
+			guestList.add(vipGuest);
```
Comment (IsisJ) [inline src/_GuestBook/GuestBook.java:36]: Interesting that you put the "VIP" here 🍦

File: src/_GuestBook/VIPGuest.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+	}
+	String name;
+	
+	public String toString() {
+		return name;
+	}
+
+	@Override
+	public String getListing() {
+		// TODO Auto-generated method stub
```
Comment (DCamaron1) [inline src/_GuestBook/VIPGuest.java]: If you're all done, go ahead and remove the "@override" and the todo comment!

File: src/_GuestBook/GuestBook.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+import javax.swing.text.JTextComponent;
+
+public class GuestBook implements ActionListener {
+	JPanel panel = new JPanel();
```
Comment (mwoguri) [inline src/_GuestBook/GuestBook.java]: These instance variables can be private.

File: src/_GuestBook/GuestBook.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+		
+		
+		if(buttonPressed == guestButton) {
+			RegularGuest regularGuest = new RegularGuest(guestNameField.getText());
+			guestList.add(regularGuest);
+			System.out.println(guestList);
+		}
+		
+		if(buttonPressed == VIPButton) {
+			VIPGuest vipGuest = new VIPGuest("VIP: " +guestNameField.getText());
```
Comment (mwoguri) [inline src/_GuestBook/GuestBook.java]: Please store just the name in the name instance variable.

File: src/_GuestBook/GuestBook.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+		}
+
+	}
+
+	public void createUI() {
+		frame.setVisible(true);
+		frame.add(panel);
+		frame.setSize(400, 200);
+		panel.add(checkInText);
+		checkInText.setText("Welcome to the Hotel California! Please check in below.");
```
Comment (mwoguri) [inline src/_GuestBook/GuestBook.java:50]: Haha!

File: src/_GuestBook/GuestBook.java [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+	}
+	
+	public static void main(String[] args) {
+		GuestBook guestbook = new GuestBook();
+		guestbook.createUI();
+	}
+	
+	public String buildNiceList(ArrayList list) {
+		return "test";
```
Comment (mwoguri) [inline src/_GuestBook/GuestBook.java]: Please implement so your display is prettier :)

File: src/_GuestBook/VIPGuest.java (hunk 8 above)
Comment (mwoguri) [inline src/_GuestBook/VIPGuest.java]: You'll have to return something here instead of `null` once you implement `buildNiceList()`!

**Review by mwoguri — APPROVED** [review]
Looks good! Please merge :)

**Pull request merged**